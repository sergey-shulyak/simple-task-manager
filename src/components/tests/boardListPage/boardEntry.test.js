import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import BoardEntry from '../../boardListPage/boardList/boardEntry';

const props = {
  id: '42',
  title: 'Sample title',
  description: 'Sample description',
  className: 'custom',
  showModal: jest.fn()
};

describe('BoardEntry', () => {
  it('Should render correctly', () => {
    const wrapper = shallow(<BoardEntry {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
