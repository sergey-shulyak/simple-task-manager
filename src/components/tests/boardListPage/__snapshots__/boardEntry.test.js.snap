// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardEntry Should render correctly 1`] = `
<Link
  className="board-entry__wrapper"
  replace={false}
  to="/boards/42"
>
  <div
    className="board-entry custom"
  >
    <div
      className="board-entry__info"
    >
      <h2
        className="board-entry__title"
      >
        Sample title
      </h2>
      <p
        className="board-entry__description"
      >
        Sample description
      </p>
    </div>
    <div
      className="board-entry__controls"
    >
      <button
        className="board-entry__edit"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-pencil"
        />
         Edit
      </button>
      <button
        className="board-entry__delete"
        onClick={[Function]}
      >
        <i
          aria-hidden="true"
          className="fa fa-trash"
        />
         Delete
      </button>
    </div>
  </div>
</Link>
`;
